File: Models/Report.cs
using System.Collections.Generic;

namespace PikeSafetyWebApp.Models
{
    public class Report : EntityBase
    {
        public string Title { get; set; }
        public long ReportTypeId { get; set; }
        public string ReportType { get; set; }
        public string ReportFields { get; set; } //stringified json object based on report type fields with additional value parameter
        public long CompanyId { get; set; }
        public string CompanyName { get; set; }
        public long SiteId { get; set; }
        public string SiteName { get; set; }
        public double CompletionPercentage { get; set; }
        public bool RequireImages { get; set; }
        public ICollection<ReportImage> Images { get; set; } = new List<ReportImage>();
    }
}

File: clientapp/src/app/models/report.ts
import { IReportField } from "./reportField";
import { IReportImage } from "./reportImage";

export interface IReport {
    id: number;
    title: string;
    reportType: string;
    reportTypeId: number;
    reportFields: IReportField[];
    companyId: number;
    companyName: string;
    siteId: number;
    siteName: string;
    createdBy: string;
    createdOn: Date;
    updatedBy: string;
    updatedOn: Date;
    isComplete: boolean;
    isActive: boolean;
    images: IReportImage[];
}